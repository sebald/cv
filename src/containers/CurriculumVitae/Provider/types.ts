import * as React from 'react';
import { CurriculumVitae } from '..';

export type CVProviderProps = {
  cv: CurriculumVitae;
  children: React.ReactNode;
};
