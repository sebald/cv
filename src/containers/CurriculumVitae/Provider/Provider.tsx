import * as React from 'react';

import { CurriculumVitae } from '..';
import { CVProviderProps } from './types';

const CurriculumVitaeContext = React.createContext<CurriculumVitae>({} as any);

// Provider
// ---------------
export const CvProvider: React.SFC<CVProviderProps> = ({ cv, children }) => (
  <CurriculumVitaeContext.Provider value={cv}>
    {children}
  </CurriculumVitaeContext.Provider>
);

// Consumer
// ---------------
export const CvConsumer = CurriculumVitaeContext.Consumer;
