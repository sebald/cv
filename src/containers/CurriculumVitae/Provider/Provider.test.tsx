import { mount } from 'enzyme';
import * as React from 'react';

import { CvConsumer, CvProvider } from './Provider';

test('render provider', () => {
  const wrapper = mount(<CvProvider cv={{} as any}>child</CvProvider>);
  expect(wrapper).toMatchSnapshot();
});

test('render consumer', () => {
  const spy = jest.fn(() => <div />);
  mount(
    <CvProvider cv={{ foo: 'bar' } as any}>
      <div>
        <CvConsumer>{spy}</CvConsumer>
      </div>
    </CvProvider>
  );

  expect(spy.mock.calls).toMatchSnapshot();
});
