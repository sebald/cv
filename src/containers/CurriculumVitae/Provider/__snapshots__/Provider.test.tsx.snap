// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render consumer 1`] = `
Array [
  Array [
    undefined,
  ],
]
`;

exports[`render provider 1`] = `
<Component
  cv={Object {}}
>
  child
</Component>
`;
